tests/utilities/test_apply_func_torchtext.py

```diff
@@ -0,0 +1,63 @@
+import torch
+import torchtext
+from torchtext.data.example import Example
+
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+
+
+def _get_torchtext_data_iterator(include_lengths=False):
+    text_field = torchtext.data.Field(sequential=True, pad_first=False,  # nosec
+                                      init_token="<s>", eos_token="</s>",  # nosec
+                                      include_lengths=include_lengths)  # nosec
+
+    example1 = Example.fromdict({"text": "a b c a c"}, {"text": ("text", text_field)})
+    example2 = Example.fromdict({"text": "b c a a"}, {"text": ("text", text_field)})
+    example3 = Example.fromdict({"text": "c b a"}, {"text": ("text", text_field)})
+
+    dataset = torchtext.data.Dataset([example1, example2, example3],
+                                     {"text": text_field}
+                                     )
+    text_field.build_vocab(dataset)
+
+    iterator = torchtext.data.Iterator(dataset, batch_size=3,
+                                       sort_key=None, device=None,
+                                       batch_size_fn=None,
+                                       train=True, repeat=False, shuffle=None,
+                                       sort=None, sort_within_batch=None)
+    return iterator, text_field
+
+
+def test_batch_move_data_to_device_torchtext_include_lengths_false():
```

Comment:
can we combine these two tests into one and parameterize? You can also parameterize the device. See how it's done in other tests if you don't know what I mean. The motivation is obviously to avoid code dupliation.

Reply:
Good suggestion. Will rewrite the test.

Reply:
@awaelchli Getting the test to work was more convoluted than I had expected. I tried to figure out why only a single device was tested on the CI machines until I realized that none of the Ubuntu machines seem to have a GPU, and the test was running on CPU only. I was able to confirm that the tests succeed on CPU and GPU using Colab with a GPU.

Do you know how to tell which CI machines have CUDA/GPU support?

The tests are now quite compact. What do you think? I appreciate your suggestions.

Reply:
I added an additional test for TPU devices and tested it on Colab. Note, getting the test using a TPU is not straight forward since it requires to install a library as described in here: https://stackoverflow.com/questions/60718831/how-to-use-torchaudio-with-torch-xla-on-google-colab-tpu .

The TPU test passed on Colab with TPU.

Reply:
well at this moment this function is not tested with TPU, only tests/models/test_tpu.py

Reply:
@borda yes I noticed that the added test was not run during CI. That is why I tested it on my own on Colab to make sure it works. I have confidence that the test is working on a TPU. Should the test on TPU stay the way it is at the moment, be always skipped, moved to tests/models/test_tpu.py, or completely  removed? I appreciate your and @awaelchli advice on this.